### DISCUSSION TITLE
confusion over where worst slacks are stored and populated

### DISCUSSION THREAD

**Discussion by cshi15** [The-OpenROAD-Project/OpenROAD#3037] [Q&A]
Hello, I'm an undergrad who has been trying to add some new functionality and thus trying to learn how the codebase works.

Right now, I am working in Resizer and trying to write a function that will print out the top failing (negative slack) paths. I was studying the code for worst slacks by studying the `worstSlack` functions in `Search.cc` and `WorstSlack.cc`, and also the `findResizeSlacks()` function in `Resizer.cc`. Looking at the `findResizeSlacks1()` function in `Resizer.cc`, it appears to populate the `worst_slack_nets_` field. However, when I try to access `worst_slack_nets_` in my function, it appears to be empty.

Is there something that I need to call before this to populate the field? Should I even be trying to access `worst_slack_nets_` at all?

**Comment by rovinski** [comment]
@openroadie @maliberty any ideas here?

@cshi15 maybe take a quick look at #3045 to see if there was anything useful there.
Also maybe if you can commit what you have to a fork then it will make it easier to understand what you have tried so far.

**Reply by openroadie** [reply]
I would suggest following the RepairSetup flow here:

https://github.com/The-OpenROAD-Project/OpenROAD/blob/954f712d52e6a672fc5f929fa3ded0b2a0901895/src/rsz/src/RepairSetup.cc#L110

It probably has everything you need.

**Comment by maliberty** [comment]
I'm guessing you need to call Resizer::resizePreamble() first.

**Reply by rovinski** [reply]
I believe @cshi15 tried both with and without the preamble and it didn't work in either case.

**Comment by maliberty** [comment]
Can you put the code on a branch somewhere?
